File: src/Helpers/file.py
class FileSubItem:
    """
    Represents a basic file with attributes like name, path, size, and parent_id.
    """

    def __init__(self, name, path, identifier, parent_id=None, size=None):
        self.name = name
        self.path = path
        self.identifier = identifier
        self.parent_id = parent_id
        self.size = size

    def __repr__(self):
        return f"FileSubItem(name={self.name}, path={self.path}, size={self.size})"


class File:
    """
    Represents a file with separate source and destination attributes.
    """

    def __init__(self, source: FileSubItem = None, destination: FileSubItem = None):
        self.source = source
        self.destination = destination

    def __repr__(self):
        return (
            f"File(Source={self.source}, "
            f"Destination={self.destination})"
        )

File: filesystem/file.go
package filesystem
import "fmt"

// File represents a basic file with attributes like name, path, size, and parentID.
type File struct {
	Name          string
	Path          string
	Identifier    string
	ParentID      string 
	Size          int64  
	LastModified  string 
	Level         int
}

// String method to provide a formatted representation of File (similar to __repr__)
func (f File) String() string {
	return fmt.Sprintf("File(name=%s, path=%s, size=%v, lastModified=%s, level=%v)", f.Name, f.Path, f.Size, f.LastModified, f.Level)
}